pkg/cmd/config/cmd.go

```diff
@@ -0,0 +1,145 @@
+package config
+
+// Copyright 2024 The KitOps Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import (
+	"context"
+	"fmt"
+	"kitops/pkg/lib/constants"
+	"kitops/pkg/output"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	shortDesc = `Manage configuration for KitOps CLI`
+	longDesc  = `Allows setting, getting, listing, and resetting configuration options for the KitOps CLI.
+
+This command provides functionality to manage configuration settings such as
+storage paths, credentials file location, CLI version, and update notification preferences.
+The configuration values can be set using specific keys, retrieved for inspection, listed,
+or reset to default values.`
+
+	example = `# Set a configuration option
+kit config set storageSubpath /path/to/storage
+
+# Get a configuration option
+kit config get storageSubpath
+
+# List all configuration options
+kit config list
+
+# Reset configuration to default values
+kit config reset`
+)
+
+func (opts *configOptions) complete(ctx context.Context, args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("no configuration key provided")
+	}
+	opts.key = args[0]
+
+	if len(args) > 1 {
+		opts.value = args[1]
+	}
+
+	configHome, ok := ctx.Value(constants.ConfigKey{}).(string)
+	if !ok {
+		return fmt.Errorf("default config path not set on command context")
+	}
+	opts.configHome = configHome
+
+	return nil
+}
+
+// ConfigCommand represents the config command
+func ConfigCommand() *cobra.Command {
+	opts := &configOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "config [set|get|list|reset] <key> [value]",
```

Comment:
If I recall, using `<` and `>` in CLI documentation breaks the auto-generated docs. Our approach is to document variable parameters with all-caps (`config [set|get|list|reset] KEY [VALUE]`)

Reply:
refactored